central/debug/service/diagnostics.go

```diff
@@ -141,6 +144,20 @@ func pullCentralClusterDiagnostics(ctx context.Context, filesC chan<- k8sintrosp
 	}
 }
 
+func getK8sInClusterConfig() (*rest.Config, error) {
+	restCfg, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	// Replacing raw IP address with kubernetes.default.svc
+	// allows for easier proxy configuration.
+	if env.ManagedCentral.BooleanSetting() {
+		port := os.Getenv("KUBERNETES_SERVICE_PORT")
```

Comment:
Is that environment variable guaranteed to be set ?
Would it make sense to parse `restCfg.Host` and replace the actual hostname with the service DNS name ?

Reply:
Yes, otherwise `rest.InClusterConfig()` would return an error